### PR TITLE
Remove orca grid

### LINKED ISSUE #983: Deprecate ORCA grid
I suggest we deprecate the ORCA / CPOM grid code

Its [not a lot](https://github.com/search?q=repo%3ACICE-Consortium%2FCICE%20orca&type=code) of code, but does helps to keep things tidy (especially in `ice_read_write.F90` ).

### PR DESCRIPTION
**Pull request by anton-seaice**
## PR checklist
- [ ] Short (1 sentence) summary of your PR: 
    Deprecate ORCA grid - closes #983 
- [ ] Developer(s): 
    @anton-seaice 
- [ ] Suggest PR reviewers from list in the column to the right.
- [ ] Please copy the PR test results link or provide a summary of testing completed below.
     
- How much do the PR code changes differ from the unmodified code? 
    - [x] bit for bit
    - [ ] different at roundoff level
    - [ ] more substantial 
- Does this PR create or have dependencies on Icepack or any other models?
    - [ ] Yes
    - [x] No
- Does this PR update the Icepack submodule?  If so, the Icepack submodule must point to a hash on Icepack's main branch.
    - [ ] Yes
    - [x] No
- Does this PR add any new test cases?
    - [ ] Yes
    - [x] No
- Is the documentation being updated? ("Documentation" includes information on the wiki or in the .rst files from doc/source/, which are used to create the online technical docs at https://readthedocs.org/projects/cice-consortium-cice/. A test build of the technical docs will be performed as part of the PR testing.)
    - [x] Yes
    - [ ] No, does the documentation need to be updated at a later time?
        - [ ] Yes
        - [ ] No 

The wiki might need updating also ?

- [ ] Please document the changes in detail, including _why_ the changes are made.  This will become part of the PR commit log.

Deprecate ORCA/CPOM grid functionality as it is not uses in CICE6. The model will abort if `orca_halogrid = .true.` or `gridtype='cpom_grid'` in the namelist.

### CODE REVIEW

**Comment by eclare108213** [comment]
@phil-blain Please confirm that this PR does not interfere with NEMO in your systems.

**Comment by anton-seaice** [comment]
I ran the base_suite and all tests pass with no regressions / results are BFB

File: cicecore/cicedyn/general/ice_init.F90 [hunk 1]
```diff
@@ -1270,7 +1271,7 @@ subroutine input_data
          endif
          abort_list = trim(abort_list)//":1"
       endif
-
+
```
Comment (phil-blain) [inline cicecore/cicedyn/general/ice_init.F90:1274]: can we loose the whitespace changes ? thanks.

**Review by phil-blain — APPROVED** [review]
I looked at the changes locally using `git show --color-moved --color-moved-ws=allow-indentation-change` and everything looks in order.

We do not use the `orca_halogrid` flag (nor the CPOM grid) so this does not impact us. 

I appreciate the code simplification:) Thanks Anton.

**Comment by apcraig** [comment]
I am testing this and #985 on derecho.  Just FYI, #985 and #986 have conflicts in 

cicecore/cicedyn/infrastructure/ice_read_write.F90
configuration/scripts/tests/base_suite.ts

I have resolved them in my sandbox, but they will need to be resolved in the PRs before we merge.  I suggest we merge one PR, then resolve the conflicts in the other PR before merging it.  The conflicts are straight-forward, although we will need to decide whether we want "short" or "medium" runlength settings on some of the tests.

File: cicecore/cicedyn/general/ice_init.F90 (hunk 1 above)
Comment (anton-seaice) [inline cicecore/cicedyn/general/ice_init.F90:1274]:
```suggestion
```

**Comment by anton-seaice** [comment]
Thanks @apcraig - sounds good. Longer is better for the runlength.

**Comment by apcraig** [comment]
Ran the quick suite on derecho with 6 compilers and all tests pass, tested done with #985.  Will merge this week, trying to merge fsd mods first.

**Review by apcraig — APPROVED** [review]

**Comment by apcraig** [comment]
@anton-seaice, can you reconcile the conflicts with just merged #985 and update the PR.  Thanks!

**Comment by anton-seaice** [comment]
This is ready to go - thanks Tony :)

**Pull request merged**